### PR TITLE
Bracket list expression for initialization without escaping

### LINKED ISSUE #1981: Add "square bracket list literal" to grammar
I'm not sure to whom to assign this. Please help.

I'm trying to update [Expressions.md](https://www.lf-lang.org/docs/handbook/expressions?target=py#collections) for Python, and I have found several problems.  Consider the following program:
```
target Python

reactor Foo(param(1, 2, 3)) {
  state x(1, 2, 3)

  reaction(startup) {=
    print("parameter param = ", self.param);
    print("type is ", type(self.param));
    print("state x = ", self.x);
    print("type is ", type(self.x));
  =}
}

main reactor {
  f = new Foo(param(3, 4, 5))
}
```
According to the documentation, `param` is supposed to be a tuple and `x` is supposed to be a list. But they are both lists. The output is:

```
parameter param =  [3, 4, 5]
type is  <class 'list'>
state x =  [1, 2, 3]
type is  <class 'list'>
```

The formatter also messes up with the above code.  It gives these warnings:

```
3 | reactor Foo(param(1, 2, 3)) {
  |                  ^ This syntax is deprecated in the Python target, use an equal sign instead of parentheses for assignment.
  |
4 |   state x(1, 2, 3)

lfc: warning: This syntax is deprecated in the Python target, use an equal sign instead of parentheses for assignment.
--> code/c/src/Junk.lf:4:10
  |
3 | reactor Foo(param(1, 2, 3)) {]
4 |   state x(1, 2, 3)
  |          ^ This syntax is deprecated in the Python target, use an equal sign instead of parentheses for assignment.
  |
5 | 

lfc: warning: This syntax is deprecated in the Python target, use an equal sign instead of parentheses for assignment.
 --> code/c/src/Junk.lf:15:20
   |
14 | main reactor {
15 |   f = new Foo(param(3, 4, 5))
   |                    ^ This syntax is deprecated in the Python target, use an equal sign instead of parentheses for assignment.
   |
16 | }

```

I try to fix the syntax, and the formatter changes it right back! (in VS code, it does this when I save the file).

If I fix the syntax in another editor to get this:

```
target Python

reactor Foo(param = (1, 2, 3)) {
  state x = (1, 2, 3)

  reaction(startup) {=
    print("parameter param = ", self.param);
    print("type is ", type(self.param));
    print("state x = ", self.x);
    print("type is ", type(self.x));
  =}
}

main reactor {
  f = new Foo(param = (3, 4, 5))
}
```

then the compilation fails:

```
> Task :cli:lff:run FAILED
lff: error: no viable alternative at input '('
--> code/c/src/Junk.lf:3:21
  |
2 | 
3 | reactor Foo(param = (1, 2, 3)) {
  |                     ^ no viable alternative at input '('
  |
4 |   state x = (1, 2, 3)

lff: error: mismatched input '2' expecting RULE_ID
--> code/c/src/Junk.lf:3:25
  |
2 | 
3 | reactor Foo(param = (1, 2, 3)) {
  |                         ^ mismatched input '2' expecting RULE_ID
  |
4 |   state x = (1, 2, 3)

lff: error: mismatched input '3' expecting RULE_ID
--> code/c/src/Junk.lf:3:28
  |
2 | 
3 | reactor Foo(param = (1, 2, 3)) {
  |                            ^ mismatched input '3' expecting RULE_ID
  |
4 |   state x = (1, 2, 3)

lff: error: extraneous input ')' expecting '{'
--> code/c/src/Junk.lf:3:30
  |
2 | 
3 | reactor Foo(param = (1, 2, 3)) {
  |                              ^ extraneous input ')' expecting '{'
  |
4 |   state x = (1, 2, 3)

lff: error: no viable alternative at input '('
--> code/c/src/Junk.lf:4:13
  |
3 | reactor Foo(param = (1, 2, 3)) {
4 |   state x = (1, 2, 3)
  |             ^ no viable alternative at input '('
  |
5 | 

lff: error: no viable alternative at input '1'
--> code/c/src/Junk.lf:4:14
  |
3 | reactor Foo(param = (1, 2, 3)) {
4 |   state x = (1, 2, 3)
  |              ^ no viable alternative at input '1'
  |
5 | 

lff: error: no viable alternative at input '2'
--> code/c/src/Junk.lf:4:17
  |
3 | reactor Foo(param = (1, 2, 3)) {
4 |   state x = (1, 2, 3)
  |                 ^ no viable alternative at input '2'
  |
5 | 

lff: error: no viable alternative at input '3'
--> code/c/src/Junk.lf:4:20
  |
3 | reactor Foo(param = (1, 2, 3)) {
4 |   state x = (1, 2, 3)
  |                    ^ no viable alternative at input '3'
  |
5 | 

lff: error: 'reaction' is a reserved keyword which cannot be used as an identifier.
--> code/c/src/Junk.lf:6:3
  |
5 | 
6 |   reaction(startup) {=
  |   ^^^^^^^^ 'reaction' is a reserved keyword which cannot be used as an identifier.
  |
7 |     print("parameter param = ", self.param);

lff: fatal error: Aborting due to 9 previous errors.

FAILURE: Build failed with an exception.
```

### PR DESCRIPTION
**Pull request by oowekyala**
Add an expression kind for Python and TS lists, and Rust arrays. 

Fix #1981 

This requires an update to the website.

### CODE REVIEW

**Comment by lhstrh** [comment]
Fantastic! This is _really_ nice, @oowekyala. It occurs to me that even if we were to aim for a clearer separation between target code and LF code, initializers are probably the places where it would remain to pay off to elegantly mix languages (i.e., without escaping)...

**Comment by lhstrh** [comment]
Looks like our `epoch` workflow doesn't deal well with forks. It's failing because it is attempting to resolve the ref `clem.bracket-list-expr` that corresponds to this PR in the `lf-lang` repo instead of your fork... I guess the workflow should take as input a string that identifies the fork and then just replace the `lingua-franca` submodule before checking out the given ref...

**Comment by lhstrh** [comment]
> Looks like our `build-epoch` workflow doesn't deal well with forks. It's failing because it is attempting to resolve the ref `clem.bracket-list-expr` that corresponds to this PR in the `lf-lang` repo instead of your fork... I guess the workflow should take as input a string that identifies the fork and then just replace the `lingua-franca` submodule before checking out the given ref...

OK, fixed this!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,7 +32,8 @@ jobs:
           files: core/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml,cli/base/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml,cli/lfc/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml,cli/lfd/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml,cli/lff/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml
 
   epoch:
-    uses: lf-lang/epoch/.github/workflows/build.yml@main
+    uses: lf-lang/epoch/.github/workflows/build.yml@build-against-fork
```
Comment (lhstrh) [inline .github/workflows/build.yml:35]: Change back to `main` after https://github.com/lf-lang/epoch/pull/53 has been merged.

Comment (lhstrh) [inline .github/workflows/build.yml:35]:
```suggestion
    uses: lf-lang/epoch/.github/workflows/build.yml@main
```

**Comment by oowekyala** [comment]
Thanks Marten for fixing the CI!

**Review by cmnrd — DISMISSED** [review]
This looks great! The only request I have, is that more Python tests should be updated. For instance `ArrayAsParamenter.lf` still uses the old syntax. There might be more.

This is beyond this PR, but we should soon remove the old `name(<expr>*)` syntax (https://github.com/lf-lang/lingua-franca/issues/1675).

**Pull request merged**